In [20]:
import numpy as np


In [21]:
np.pi

3.141592653589793

In [22]:
data=np.random.random(size=10)*2*np.pi-np.pi#-pi_+pi

In [23]:
np.sign(data)

array([-1.,  1., -1., -1., -1.,  1.,  1., -1., -1.,  1.])

`np.sign(np.pi)` 用于判断 `np.pi`（即圆周率 π，约等于 3.14159）的符号。`np.sign` 函数会返回：

- 1，如果输入为正数
- 0，如果输入为零
- -1，如果输入为负数

由于 π 是正数，因此 `np.sign(np.pi)` 的结果为 1。

In [24]:
np.sin(data)

array([-0.95131083,  0.94838813, -0.37795656, -0.98461313, -0.06589221,
        0.0867445 ,  0.64310576, -0.76692197, -0.14287256,  0.3294244 ])

In [25]:
np.sin(90),np.sin(np.pi/2)#用弧度

(0.8939966636005579, 1.0)

In [26]:
data=np.random.random(size=10)

In [27]:
np.around(data,3)#保留3位小数

array([0.42 , 0.43 , 0.398, 0.888, 0.396, 0.081, 0.954, 0.944, 0.328,
       0.228])

In [28]:
a1=np.random.randint(0,10,size=(5,3))
b1=np.random.randint(0,10,size=3)

In [29]:
np.add(a1, b1)

array([[10, 10, 13],
       [10, 11, 11],
       [ 6, 11, 12],
       [ 9, 15, 10],
       [10, 12,  7]])

In [30]:
np.power(3,2)

9

In [31]:
data=np.random.randint(0,10,size=5)
data

array([4, 2, 2, 0, 9])

In [32]:
data_s=np.power(data,2)


In [33]:
np.power(data_s,1/2)

array([4., 2., 2., 0., 9.])

In [34]:
np.mod(5,2)#余数

1

In [35]:
np.mod(data,2)

array([0, 0, 0, 0, 1])

In [36]:
np.e

2.718281828459045

In [37]:
np.log10(100)

2.0